src/couch/src/couch_httpd_auth.erl

```diff
@@ -308,15 +308,26 @@ cookie_authentication_handler(#httpd{mochi_req = MochiReq} = Req, AuthModule) ->
                         {ok, UserProps, _AuthCtx} ->
                             UserSalt = couch_util:get_value(<<"salt">>, UserProps, <<"">>),
                             FullSecret = <<Secret/binary, UserSalt/binary>>,
-                            ExpectedHash = couch_util:hmac(sha, FullSecret, User ++ ":" ++ TimeStr),
+                            CalculatedHashes = lists:map(
```

Comment:
this will cause us to hash each cookie with every algorithm each time, which is wasteful. instead we want to stop at the first successful match.

Reply:
Added a predicate function to test the hash in combination with lists:any/2